### PR TITLE
feat: implement drop multiple tables

### LINKED ISSUE #4020: Support to drop multiple tables
### What problem does the new feature solve?

Drop multiple tables in the single statement

### What does the feature do?

To implement the `DROP [TEMPORARY] TABLE [IF EXISTS] tbl_name [, tbl_name]`

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by sarailQAQ**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
close #4020 
## What's changed and what's your intention?

Implemet dropping multiple tables in the single statement.

If any tables do not exist without IF EXISTS, the statement fails with an error and don`t make any change, referencing MySQL.

It works for SQL requests, not ready for gRPC.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

**Comment by WenyXu** [comment]
Would you like to add some sqlness tests in https://github.com/GreptimeTeam/greptimedb/blob/45fee948e9cb02ecbd9f224385d16f87e7c1860d/tests/cases/standalone/common/drop/drop_table.sql#L1-L13

Including the cases of success, partial success, and failure

**Comment by WenyXu** [comment]
The Rustfmt and Clippy checks are failed, run `make fmt` and `make clippy` will help you to fix them

**Comment by sarailQAQ** [comment]
image
It seems runner does not have enough space? How to fix it?

**Comment by WenyXu** [comment]
> image It seems runner does not have enough space? How to fix it?

We are still trying to fix it #4081

**Comment by WenyXu** [comment]
Hi @sarailQAQ let’s rebase this PR, the CI should be fixed

**Review by evenyag — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]

File: src/operator/src/statement/ddl.rs [hunk 1]
```diff
@@ -643,42 +643,59 @@ impl StatementExecutor {
...
+                continue;
+            } else {
+                return TableNotFoundSnafu {
+                    table_name: table_name.to_string(),
+                }
+                .fail();
+            }
+        }
+
+        for (table_name, table_id) in table_names.iter().zip(tables.iter()) {
```
Comment (WenyXu) [inline src/operator/src/statement/ddl.rs:682]: We can consume the `table_names`.
```suggestion
        for (table_name, table_id) in table_names.into_iter().zip(tables.iter()) {
```

File: src/operator/src/statement/ddl.rs [hunk 2]
```diff
@@ -643,42 +643,59 @@ impl StatementExecutor {
...
                 .await?;
 
             // Invalidates local cache ASAP.
             self.cache_invalidator
                 .invalidate(
                     &Context::default(),
                     &[
-                        CacheIdent::TableId(table_id),
+                        CacheIdent::TableId(*table_id),
                         CacheIdent::TableName(table_name.clone()),
```
Comment (WenyXu) [inline src/operator/src/statement/ddl.rs:692]: Can we avoid the `clone` here?

Comment (sarailQAQ) [reply src/operator/src/statement/ddl.rs:692]: For now, we can\`t move ownership from `&[TableName]`, maybe `Vec<TableName>` is ok.

In addition, the original implementation of `fn drop_table` here uses `table_name.clone()`:
https://github.com/GreptimeTeam/greptimedb/blob/a6269397c8924dbf9914ce13d81eded74a8c8f68/src/operator/src/statement/ddl.rs#L666
So I just follow it, for I`m not sure it is ok to remove it.

**Pull request merged**